cloud/scope/vpc.go

```diff
@@ -54,11 +54,19 @@ func validateVPCScopeParams(params VPCScopeParams) error {
 
 // NewVPCScope creates a new Scope from the supplied parameters.
 // This is meant to be called for each reconcile iteration.
-func NewVPCScope(apiKey string, params VPCScopeParams) (*VPCScope, error) {
+func NewVPCScope(ctx context.Context, apiKey string, params VPCScopeParams) (*VPCScope, error) {
 	if err := validateVPCScopeParams(params); err != nil {
 		return nil, err
 	}
 
+	// Override the controller credentials with ones from the VPC's Secret reference (if supplied).
+	if params.LinodeVPC.Spec.CredentialsRef != nil {
```

Comment:
is `LinodeCluster` in the scope of this as well? I think it would be nice if it could just be set at the linodeCluster level and be inherited by the other resources under it.  we can still keep this part in, but just default to the cluster if this isn't specified possibly?

Reply:
Please see the Reviewer Notes in the Description. I'm happy to discuss if we want to take on a bigger refactor of our reconciliation logic though.

Reply:
ah right, thanks for pointing me back at that I had read it and forgot while reviewing. happy to chat through this but given we are creating VPCs for lkeClusters I think being able to just define it there if possible would be preferable as the default

Reply:
Sounds good! Since the ObjectStorageBucket reconciler (#154) also uses this pattern, I think we might want this be a bigger conversation about redesigning how the various controllers work together (and what information controllers should wait for/need when creating the backing Linode resource).